ui/app/pages/confirmation/confirmation.js

```diff
@@ -0,0 +1,243 @@
+    if (
+      pendingConfirmations.length === 0 &&
+      (environmentType === ENVIRONMENT_TYPE_FULLSCREEN ||
+        environmentType === ENVIRONMENT_TYPE_POPUP)
+    ) {
+      history.push(DEFAULT_ROUTE);
+    } else if (pendingConfirmations.length <= currentPendingConfirmation) {
+      setCurrentPendingConfirmation(pendingConfirmations.length - 1);
+    }
+  }, [pendingConfirmations, history, currentPendingConfirmation]);
+  if (!pendingConfirmation) {
+    return null;
+  }
+
+  return (
+    <div className="confirmation-page">
+      {pendingConfirmations.length > 1 && (
+        <div className="confirmation-page__navigation">
+          <p>
+            {currentPendingConfirmation + 1} of {pendingConfirmations.length}{' '}
+            pending
+          </p>
+          {currentPendingConfirmation > 0 && (
+            <button
+              className="confirmation-page__navigation-button"
+              onClick={() =>
+                setCurrentPendingConfirmation(currentPendingConfirmation - 1)
+              }
+            >
+              <i className="fas fa-chevron-left"></i>
+            </button>
+          )}
+          <button
+            className="confirmation-page__navigation-button"
+            disabled={
+              currentPendingConfirmation + 1 === pendingConfirmations.length
+            }
+            onClick={() =>
+              setCurrentPendingConfirmation(currentPendingConfirmation + 1)
+            }
+          >
+            <i className="fas fa-chevron-right"></i>
+          </button>
+        </div>
+      )}
+      <div className="confirmation-page__content">
+        <Box justifyContent="center">
+          <NetworkDisplay
+            colored={false}
+            indicatorSize={SIZES.XS}
+            labelProps={{ color: COLORS.BLACK }}
+          />
+        </Box>
+        <Box justifyContent="center" padding={[1, 4, 4]}>
+          <Chip
+            label={stripHttpSchemes(originMetadata.origin)}
+            labelColor="gray"
+            leftIcon={
+              <SiteIcon
+                icon={originMetadata.icon}
+                iconName={originMetadata.name}
+                size={32}
+              />
+            }
+          />
+        </Box>
+        <MetaMaskTemplateRenderer sections={templatedValues.content} />
+      </div>
+      <ConfirmationFooter
+        alerts={
+          alertState[pendingConfirmation.id] &&
+          Object.values(alertState[pendingConfirmation.id])
+            .filter((alert) => alert.dismissed === false)
+            .map((alert, idx, filtered) => (
+              <Callout
+                key={alert.id}
+                severity={alert.severity}
+                dismiss={() => dismissAlert(alert.id)}
+                isFirst={idx === 0}
+                isLast={idx === filtered.length - 1}
+                isMultiple={filtered.length > 1}
+              >
+                {React.isValidElement(alert.content) ? (
```

Comment:
on a call we suggested performing this check `React.isValidElement` or a more direct check (`typeof string`) inside `MetaMaskTemplateRenderer`

Reply:
Resolved in latest commit